File: ProgramingBasics/AdvancedLoops/07.GreatestCommonDivisor/src/GreatestCommonDivisor.java
import java.util.Scanner;

public class GreatestCommonDivisor {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int a = Integer.parseInt(scanner.nextLine());
        int b = Integer.parseInt(scanner.nextLine());

        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }

        System.out.println(a);
    }
}

File: ProgramingBasics/AdvancedLoops/07.GreatestCommonDivisor/07.GreatestCommonDivisor/GreatestCommonDivisor.cs
using System;

namespace _07.GreatestCommonDivisor
{
    public class GreatestCommonDivisor
    {
        public static void Main(string[] args)
        {
            var a = int.Parse(Console.ReadLine());
            var b = int.Parse(Console.ReadLine());

            while (b != 0)
            {
                var temp = b;
                b = a % b;
                a = temp;
            }

            Console.WriteLine(a);
        }
    }
}
